src/sources/datadog_agent/metrics.rs

```diff
@@ -385,9 +386,12 @@ fn into_vector_metric(
 ) -> Vec<Event> {
     let mut tags = into_metric_tags(dd_metric.tags.unwrap_or_default());
 
-    dd_metric
-        .host
-        .and_then(|host| tags.replace(log_schema().host_key().to_owned(), host));
+    if let Some(key) = log_schema().host_key() {
```

Comment:
Ultimately not due to your change, but it's very weird we're using the _log_ schema for metric events here.

Reply:
Feel free to cc someone who might have context on this.

Reply:
@jszwedko what do you think? This strikes me as metrics are also in need of semantic meanings?

Reply:
From what I can tell this question seems unrelated to this PR so I would probably just defer for now. We could raise it with Nathan when he's back.

Reply:
Yep, didn't mean to imply it was a today problem - sorry if that wasn't clear!